File: backend/models/lead.py
import uuid
from sqlalchemy import Column, String, Float, DateTime, Text, ForeignKey
from sqlalchemy.dialects.mssql import UNIQUEIDENTIFIER
from sqlalchemy.sql import func
from data.connections.database import Base


class Lead(Base):
    __tablename__ = "leads"
    __table_args__ = {"schema": "prospector"}

    id = Column(UNIQUEIDENTIFIER, primary_key=True, default=uuid.uuid4)
    tenant_id = Column(String(100), nullable=False, default="prospector-qc")
    prospection_id = Column(UNIQUEIDENTIFIER, ForeignKey("prospector.prospections.id"), nullable=False)
    nome_empresa = Column(String(300), nullable=False)
    segmento = Column(String(200))
    website = Column(String(500))
    descricao = Column(Text)
    score = Column(Float, default=0)
    status = Column(String(50), nullable=False, default="novo")
    criado_em = Column(DateTime, nullable=False, server_default=func.now())

File: frontend/src/lib/dtos/lead.ts
export interface LeadDetalhe {
	id: string;
	prospection_id: string;
	nome_empresa: string;
	segmento: string | null;
	website: string | null;
	descricao: string | null;
	score: number;
	status: string;
	contatos: Contato[];
	criado_em: string;
}

export interface Contato {
	id: string;
	nome: string | null;
	cargo: string | null;
	email: string | null;
	telefone: string | null;
	linkedin: string | null;
}
